name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/actions/RevertAccountAction.java

```diff
@@ -0,0 +1,41 @@
+package name.abuchen.portfolio.ui.views.actions;
+
+import org.eclipse.jface.action.Action;
+
+import name.abuchen.portfolio.model.AccountTransaction;
+import name.abuchen.portfolio.model.Client;
+import name.abuchen.portfolio.model.TransactionPair;
+
+public class RevertAccountAction extends Action
+{
+    private final Client client;
+    private final TransactionPair<AccountTransaction> transaction;
+
+    public RevertAccountAction(Client client, TransactionPair<AccountTransaction> transaction)
+    {
+        this.client = client;
+        this.transaction = transaction;
+
+        if (transaction.getTransaction().getType() != AccountTransaction.Type.DEPOSIT
+                        && transaction.getTransaction().getType() != AccountTransaction.Type.DIVIDENDS
```

Comment:
At least for DIVIDENDS there is no sibling defined. Would this work?

Plus: dividends can have taxes, other transactions not necessarily.

I also have the feeling we need some more tests here. For example, I want to add taxes to interest transactions. But does that apply to interest_charge as well? And probably the other way around, right. Taxes are not removed, but actually "paid" back. For now it is okay, but in the future one always has to keep in mind to change those "revert" actions.

Reply:
Valid point, the Units are not covered by this code at this point.
Regarding the sibling of DIVIDEND: it seems like I introduced DIVIDEND_CHARGE in my code in March 2018...